### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: src/atom.rs [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // See the COPYRIGHT file at the top-level directory of this distribution.
 // Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
 
+//! Atoms — Functions to manipulate properties on windows
```
Comment (gkoz) [inline src/atom.rs]: We probably won't publish the modules, only reexport types from them so this kind of annotation seems better suited to the structs not modules.

Comment (GuillaumeGomez) [inline src/atom.rs]: I was thinking about. I'll move the doc to the type.

**Comment by GuillaumeGomez** [comment]
cc @gkoz

File: src/device_manager.rs [hunk 2]
```diff
@@ -11,6 +9,103 @@ use display::Display;
...
+/// Virtual devices are always paired, so there is a keyboard device for every pointer device.
+/// Associations between devices may be inspected through Device::get_associated_device().
+///
+/// There may be several virtual devices, and several physical devices could be controlling each of
+/// these virtual devices. Physical devices may also be “floating”, which means they are not attached
+/// to any virtual device.
+///
+/// #Master and slave devices
+///
+/// ``ìgnore
```
Comment (gkoz) [inline src/device_manager.rs]: Does this part render correctly? The syntax looks weird: two backticks never terminated?

File: src/device.rs [hunk 3]
```diff
@@ -22,68 +21,158 @@ impl types::StaticType for Device {
...
+    ///     gchar *path;
+    /// 
+    ///     let vendor = device.get_vendor_id().unwrap();
+    ///     let product = device.get_product_id().unwrap();
+    /// 
+    ///     let path = format!("/org/example/app/devices/{}:{}/", vendor, product);
+    ///     g_settings_new_with_path(DEVICE_SCHEMA, path);
+    /// }
+    /// ```
+    #[cfg(feature = "gdk_3_16")]
```
Comment (gkoz) [inline src/device.rs]: Looks like these need a `s/feature = "([^"]+)"/$1/`.

File: src/device_manager.rs (hunk 2 above)
Comment (GuillaumeGomez) [inline src/device_manager.rs]: I see what happened, it became another character.

File: src/device.rs (hunk 3 above)
Comment (GuillaumeGomez) [inline src/device.rs]: What ? I don't understand your comment.

Comment (gkoz) [inline src/device.rs]: The current syntax is `cfg(gdk_3_16)` and that regex is how I'd do the conversion.

Comment (GuillaumeGomez) [inline src/device.rs]: Oh right. I forgot we changed the system !

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by gkoz** [comment]
Looks good!

File: src/device_manager.rs [hunk 4]
```diff
@@ -11,6 +9,103 @@ use display::Display;
...
+///
+/// There may be several virtual devices, and several physical devices could be controlling each of
+/// these virtual devices. Physical devices may also be “floating”, which means they are not attached
+/// to any virtual device.
+///
+/// #Master and slave devices
+///
+/// ```ignore
+/// carlos@sacarino:~$ xinput list
+/// ⎡ Virtual core pointer                          id=2    [master pointer  (3)]
```
Comment (gkoz) [inline src/device_manager.rs]: `rustdoc` chokes on this :(

**Pull request merged**